modules/navier_stokes/include/materials/NSFVPump.h

```diff
@@ -0,0 +1,67 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "FunctorMaterial.h"
+
+/**
+ * This is the material class used to compute phase averaged properties of mixtures
+ */
+class NSFVPump : public FunctorMaterial
+{
+public:
+  static InputParameters validParams();
+
+  NSFVPump(const InputParameters & parameters);
+
+protected:
+  /// Function providing the pressure head
+  /// The pressure head should be provided as head (in meters) vs. flow rate (in m3/s)
+  const Function * _pressure_head_function;
+
+  /// Rated transversal area of the pump
+  const Moose::Functor<Real> & _area_rated;
+
+  /// Rated volume of the pump
+  const Moose::Functor<Real> & _volume_rated;
```

Comment:
They could be in case we want to introduce a space function or a volume-smoothed area change in the pump. I find it more general this way. What do you think?

Reply:
works for me